022-custom-smtp-and-email-templates.md

```diff
@@ -0,0 +1,167 @@
+# Custom SMTP and Email/SMS Templates Support
+
+- Implementation Owner: @lohanidamodar
+- Start Date: 2023-03-06
+- Target Date: 2023-03-18
+- Appwrite Issue: [https://github.com/appwrite/appwrite/issues/4664](https://github.com/appwrite/appwrite/issues/4664)
+
+## Summary
+
+[summary]: #summary
+
+Support custom SMTP configuration per project and support customizing the email and sms templates. This allows users to set project specific templates as required by the project. Custom email templates will only be supported if the custom SMTP is enabled.
+
+## Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+**What problem are you trying to solve?**
+
+Right now the whole Appwrite system uses a single global SMTP configuration set via environment variables, this means all the projects will use the same email configuration. As well there is no way for user to customize the default email templates. As Appwrite supports multiple projects, which may have different needs for emailing. It is even more crucial in the cloud to support custom email templating as each individual users will have their own need for the project and may want to use their own emailing system to send emails.
+
+## Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+### API Endpoints
+
+For this we will introduce various endpoints
+
+**PATCH /v1/projects/:projectId/smtp**       - updates smtp configuration for the project
+
+It will accept following parameters
+
+- **enabled** : boolean - **required** - whether or not the SMTP configuration is enabled
+- **sender** : string - SMTP sender email
+- **host** : string - SMTP server host name address
+- **port** : string - SMTP server tcp port
+- **secure** : string - SMTP secure protocol (empty or `tls`)
+- **username** : string - SMTP server user name
+- **password** : string - SMTP server password
+
+**PATCH /v1/projects/:projectId/template/sms/:type/:locale** - updates custom SMS templates
+
+- **type** : string - required - template type
+- **locale** : string - required (eg: en_us) - locale of the template
```

Comment:
@eldadfux I'm having problem with the full locale `en_us` our locale library only accepts `en`. What's the purpose of having this full?

Reply:
There are differences between en_us and en_uk, some might want to be more accurate.